### PR TITLE
Add orientation and config options

### PR DESCRIPTION
**Pull request by pointW**
1. Add random orientation when generating object
2. Fix rotation bug when moving arm in vrep
3. Add port option in vrep config
4. Add 'action_sequence' in all env config to specify action format
5. Speed up vrep by only restart simulator when not valid

### CODE REVIEW

File: envs/block_picking_env.py [hunk 1]
```diff
@@ -10,27 +10,40 @@ class BlockPickingEnv(simulator_base_env):
...
       super(BlockPickingEnv, self).reset()
 
-      self.block = self._generateObjects(0, 1)[0]
+      self.block = self._generateShapes(0, 1, random_orientation=True)[0]
 
       return self._getObservation()
 
     def _checkTermination(self):
       ''''''
-      return self._isObjectHeld(self.block)
```
Comment (ColinKohler) [inline envs/block_picking_env.py]: Can you pull the code here for PyBullet/Vrep into a function inside their base envs with the same name as the numpy env so this call will just go back to: return slf.isHolding()

File: envs/block_picking_env.py [hunk 2]
```diff
@@ -10,27 +10,40 @@ class BlockPickingEnv(simulator_base_env):
...
+        return self._isHolding()
+      else:
+        block_position = self._getObjectPosition(self.block)
+        rest_pose = self._getRestPoseMatrix()
+        # print('{} > {}'.format(block_position[2], self.rest_pose[0][2] - 0.25))
+        return block_position[2] > rest_pose[2,-1] - 0.25
+
+    def _getObservation(self):
+      state, obs = super(BlockPickingEnv, self)._getObservation()
+      return 0, obs
```
Comment (ColinKohler) [inline envs/block_picking_env.py]: We should probably do some cleanup here to make less dumb.

File: envs/numpy_env.py [hunk 3]
```diff
@@ -44,28 +45,30 @@ def step(self, action):
 
     return obs, reward, done
 
-  def _pick(self, x, y, rot):
+  def _pick(self, x, y, z, rot):
     ''''''
     if self._isHolding():
       return self.held_object
```
Comment (ColinKohler) [inline envs/numpy_env.py]: I think I wrote this but it seems a bit weird so we might wana change it?

File: envs/numpy_env.py [hunk 4]
```diff
@@ -44,28 +45,30 @@ def step(self, action):
...
 
-  def _place(self, x, y, rot):
+  def _place(self, x, y, z, rot):
     ''''''
     pass
 
   def _getObservation(self):
     ''''''
+    for o in self.objects:
+      o.addToHeightmap(self.heightmap)
```
Comment (ColinKohler) [inline envs/numpy_env.py]: Does this re-add objects to the heightmap multiple times? Seems a bit weird unless we are reseting the height map to zeros at every time step.

File: envs/block_picking_env.py (hunk 1 above)
Comment (pointW) [reply envs/block_picking_env.py]: Please the new version

File: envs/numpy_env.py (hunk 4 above)
Comment (pointW) [reply envs/numpy_env.py]: Yes, I fixed this

**Comment by ColinKohler** [comment]
This all looks fine. If you fix zee merge conflicts we can merge away.

**Pull request merged**